portfolio/src/main/webapp/script.js

```diff
@@ -63,14 +63,42 @@ function addRandomMusical() {
 }
 
 /**
- * Add a random goodbye to the page.
+ * Adds a random goodbye to the page.
  */
-async function getRandomGoodbye() {
-  const response = await fetch('/data');
-  const goodbye = await response.text();
+function addRandomGoodbye() {
+  const goodbyes =
+      ['Goodbye!', 
+      'Au revoir!', 
+      'Ciao', 'Sayonara', 'Aloha',
+      'Arrivederci', 'Bon Voyage'];
+
+  // Pick a random goodbye.
+  const goodbye = goodbyes[Math.floor(Math.random() * goodbyes.length)];
+  // Add it to the page.
   document.getElementById('goodbye-container').innerText = goodbye;
 }
 
+/**
+ * Fetches comments from the servers and adds them to the DOM.
+ */
+function getComments() {
+  fetch('/data').then(response => response.json()).then((comments) => {
```

Comment:
is comments guaranteed to be a list by the caller, or can it be undef/null?

Reply:
comments is guaranteed to be a list by the caller since the comments ArrayList<>() is always initialized in the JavaServlet doGet() function (it could be empty however)